lib/formatters/orgsGetDetails.js

```diff
@@ -0,0 +1,227 @@
+ //    primary_first_name: 'Sally',
+ //    org_tag_name: 'UK',
+ //    org_tag_id: 27,
+ //    challenge_title: 'Challenge Number 3',
+ //    challenge_tag_name: 'Waste to energy',
+ //    challenge_tag_id: 49,
+ //    challenge_id: 3
+ //  }, {
+ //    org_id: 1,
+ //    org_name: 'Apple AAAA',
+ //    primary_id: 3,
+ //    primary_first_name: 'Sally',
+ //    org_tag_name: 'UK',
+ //    org_tag_id: 27,
+ //    challenge_title: 'Challenge Number 3',
+ //    challenge_tag_name: 'Buildings design',
+ //    challenge_tag_id: 105,
+ //    challenge_id: 3
+ //  }, {
+ //    org_id: 1,
+ //    org_name: 'Apple AAAA',
+ //    primary_id: 3,
+ //    primary_first_name: 'Sally',
+ //    org_tag_name: 'UK',
+ //    org_tag_id: 27,
+ //    challenge_title: 'Challenge Number 3',
+ //    challenge_tag_name: 'Fertiliser',
+ //    challenge_tag_id: 85,
+ //    challenge_id: 3
+ //  } ]
+ //
+
+
+
+  // Looking for something of the following form:
+  // { org: {id, name, mission, url, tags: [{tag_name, tag_id}, ...]}
+  // , primary: {id, first_name, }
+  // , challenges: [{id, title, tags: [{tag_name, tag_id}, ...]}]
+  // }
+
+/*
+  {
+    "org": {
+      "id": 1,
+      "name": "Apple AAAA",
+      "logo_url": "https://www.google.co.uk/imgres?iitter.com%2Fcirculareconomy&docid=LnflHf1c&uact=8",
+      "mission_statement": "Change the economy"
+    },
+    "primary": {
+      "first_name": "Sally",
+      "last_name": "Robbins",
+      "id": "07111111111",
+      "email": "sa@ro.co",
+      "job_title": "Athlete"
+    },
+    "challenges": [
+      {
+        "id": 2,
+        "title": "Challenge Number 2",
+        "tags": [
+          {
+            "tag_id": 2,
+            "tag_name": "Corporate"
+          }
+        ]
+      },
+      {
+        "id": 3,
+        "title": "Challenge Number 3",
+        "tags": [
+          {
+            "tag_id": 49,
+            "tag_name": "Waste to energy"
+          },
+          {
+            "tag_id": 105,
+            "tag_name": "Buildings design"
+          },
+          {
+            "tag_id": 85,
+            "tag_name": "Fertiliser"
+          }
+        ]
+      }
+    ]
+  }
+*/
+
+
+module.exports = function (rows) {
+  var org = {}; // { id, name }
+  var organisation_tags = {}; // {tag_id: {tagobj}, tag_id: {} ... }
+  var challenge_tags = {}; // {challengeId: {tag_id: {tagobj}, tag_id: {tagobj} }, ... };
+  var challenge_tags_order = {}; // {challengeId: {tag_id: {tagobj}, tag_id: {tagobj} }, ... };
+  var primary = {};
+  var challenges = {}; // { challenge_id: {id, title, tags: [ID, ID, ID]}, challenge_id: {id, title, tags: [ID, ID, ID]}}
+  var challenges_order = [];
+  rows.forEach(function (row) {
+    // if this is the first row, create the org and primary objects
+    if (!org.id) {
+      org.id = row.org_id;
+      org.name = row.org_name;
+      org.logo_url = row.org_logo_url;
+      org.mission_statement = row.org_mission_statement;
+      primary.first_name = row.primary_first_name;
```

Comment:
if an organisation had no primary user attached to this, would `row.primary_first_name` (and the other primary related keys in the row object) just return `null`?